pkg/schemalog/schema_diff.go

```diff
@@ -150,6 +163,50 @@ func computeTableDiff(old, new *Table) *TableDiff {
 		}
 	}
 
+	matchRenamedIndexes(diff)
```

Comment:
Would it make sense to deal with renames (and other `ALTER INDEX` operations) by checking during the equality comparison on line 154? We could just check if they're not equal and the description starts with `ALTER INDEX` then we can add it as a `IndexesChanged` with the description sql to be applied directly. 

I think it would remove all the complexity from the `matchRenamedIndexes` and also remove the need for crafting the sql in the [ddl adapter](https://github.com/xataio/pgstream/pull/609/commits/685a2221468d4d6474a0dec2c2df77280277fd67#diff-f44975e58ccf9594a166631b719cfdc93917eef008eef55f0133f386d7bda887R443). It would also mean we don't only cover renames but other alter index operations.